### DISCUSSION TITLE
Allow set folder for files on upload.

### DISCUSSION THREAD

**Discussion by dimitrov-adrian** [directus/directus#3022] [Feature Requests]
Folders are good addition but will be better if we can set destination folder for a file field, so for newly uploaded files they will be placed in folder.

**Comment by benhaynes** [comment]
I 100% agree with this. I think it would be nice to allow setting a default folder for the field/interface, maybe a default folder for the system (goes to root if path fails), and perhaps a folder option per storage adapter too.

**Comment by darioguarascio** [comment]
For local storage, it is also a filesystem issue. If you have thousands of files under the same folder it might create problems in the long run.
Ideally, adding a parameter with destination folder, plus a `mkdir -p` before writing to that folder.

**Comment by joelarula** [comment]
I guess my RFC suites to this thread. When creating storage asset presets i did found that all the presets derivative images are stored on same root filesystem folder as uploaded original. I would like to organize preset thumbnails into subfolders and further organize assets by the collection->field association. It would enable more atomic preset and field handling.  Lets consider for example field | preset delete, update operations. Looking from current db model it looks like quite significant change to introduce. Backwards compatibility and fs issues considered

**Comment by ghost** [comment]
Could I add to this one, when a user uploads an avatar can the be set to store in a subfolder too?

**Comment by connorwinston** [comment]
You are able to set the default file upload folder in project settings!
